projects/graph/social/social.py

```diff
@@ -45,10 +47,19 @@ def populateGraph(self, numUsers, avgFriendships):
         self.users = {}
         self.friendships = {}
         # !!!! IMPLEMENT ME
-
         # Add users
-
+        for num in range(numUsers):
+            self.addUser(f"User {num}")
         # Create friendships
+        friend_list = []
+        for i in range(numUsers):
+            for j in range(numUsers):
+                if i != j and i < j:
+                    friend_list += [[i, j]]
+        random.shuffle(friend_list)
+        friend_list = friend_list[:10]
```

Comment:
Hmm interesting. Could you try to make the relationship 1:1?

Reply:
But carry on if it works for now. You can always refactor later.

Reply:
> 
> 
> Hmm interesting. Could you try to make the relationship 1:1?

What do you mean by a 1:1 relationship?

Reply:
Why would you have an excess number of friend nodes? the friend list only contains the connections It's the edges of the vertices that is your userID value. That's what I see from how you implemented. So it will grow with respect to how many connections each user has right. You can expect not every user will be friends with everyone, but the friendships should be bidirectional. Because it doesn't seem like to me that you're containing the friend of a friend relationships in friend_list.

Reply:
My only concern is that this will cause errors when you try to look at the degrees of seperation.